lib/rspec/core/memoized_helpers.rb

```diff
@@ -123,14 +123,28 @@ def is_expected
 
       # @private
       def __memoized
-        @__memoized ||= {}
+        @__memoized ||= Memoized.new
```

Comment:
Yeah, that seems distinctly possible. I doubt there's a way to get a test to show this, but I could move it to initialize. The other possibility is to set it as soon as we know we need it, which would work with `let`, but I think there's other use cases which we might not know until the test has begun executing, at which point it could be too late.